### PR TITLE
Medium difficulty

### PR DESCRIPTION
**Pull request by tejashrighogare**

### CODE REVIEW

File: Medium_difficulty/taxSalary.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import java.util.Scanner;
+
```
Comment (rohitlamsonge66) [inline Medium_difficulty/taxSalary.java]: Program logic is incorrect. Try to under stand the question properly.

File: Medium_difficulty/taxSalary.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class taxSalary {
+    public static void taxSalaryCount(double salary) {
+        double tax = 0;
+        if (salary <= 500000)
+            tax = 0;
+        else if (salary > 500000 && salary <= 700000)
+            tax = 0.1 * (salary - 500000);
+        else if (salary > 700000 && salary <= 1000000)
```
Comment (rohitlamsonge66) [inline Medium_difficulty/taxSalary.java]: try to use switch case instead of multiple if statement.

File: Medium_difficulty/taxSalary.java (hunk 1 above)
Comment (tejashrighogare) [reply Medium_difficulty/taxSalary.java]: Okay, I will update code.

File: Medium_difficulty/taxSalary.java (hunk 2 above)
Comment (tejashrighogare) [reply Medium_difficulty/taxSalary.java]: Yes

File: Medium_difficulty/swapPrimeNo.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+public class swapPrimeNo {
+    public static void main(String[] args) {
+        int[] someArray = {1, 2, 3, 7};
+        SwapPrimeNumber(someArray);
+    }
+    public static void SwapPrimeNumber(int[] X) {
+        for (int i = 0; i <= X.length; i++) {
```
Comment (rohitlamsonge66) [inline Medium_difficulty/swapPrimeNo.java]: Use the proper variable name in program which is understandable

Comment (tejashrighogare) [reply Medium_difficulty/swapPrimeNo.java]: Yes

File: Medium_difficulty/Patterns.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    public static void patternNo() {
+        for (int i = 1; i <= 4; i++) {
+            int n = 4;
+            for (int j = 1; j <= n - i; j++) {
+                System.out.print(" ");
+            }
+            for (int k = i; k >= 1; k--) {
+                System.out.print(k);
+            }
```
Comment (rohitlamsonge66) [inline Medium_difficulty/Patterns.java]: spacing is not proper in the output of first vale of prymid.

Comment (rohitlamsonge66) [reply Medium_difficulty/Patterns.java]: expected 
    1
  212
 32123
4321234
but actual is 
1
  212
 32123
4321234

File: Medium_difficulty/swapPrimeNo.java (hunk 3 above)
Comment (tejashrighogare) [reply Medium_difficulty/swapPrimeNo.java]: I have resolved

File: Medium_difficulty/taxSalary.java (hunk 1 above)
Comment (tejashrighogare) [reply Medium_difficulty/taxSalary.java]: I have resolved

File: NewPrograms/JavaExceptionExample.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+		       int data=100/0;  
+		   }catch(ArithmeticException e)
+		   {
+			   System.out.println(e);
+			   }  
+		    
+		   System.out.println("rest of the code...");  
+		  }  
+		
+
```
Comment (rohitlamsonge66) [inline NewPrograms/JavaExceptionExample.java]: Remove not required space.

File: NewPrograms/QueueClass.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
+package Occurances;
+
+public class QueueClass {
+	public static void main(String[] args) {
+		Queue q = new Queue(4);
```
Comment (rohitlamsonge66) [inline NewPrograms/QueueClass.java]: Give the meaning full variable name

File: NewPrograms/QueueClass.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	private static int queue[];
+	Queue(int size) {
+		front = rear = 0;
+		capacity = size;
+		queue = new int[capacity];
+	}
+
+	void queueEnqueue(int item) {
+		if (capacity == rear) {
+			System.out.printf("\nQueue is full\n");
```
Comment (rohitlamsonge66) [inline NewPrograms/QueueClass.java]: Remove unneccesary space.

File: NewPrograms/funOverloading.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    // This function takes two double parameters
+    public double Add(double x, double y) {
+        return (x + y);
+    }
+}
+
+
+
+
```
Comment (rohitlamsonge66) [inline NewPrograms/funOverloading.java]: Remove not required space.

File: NewPrograms/interfaceJava.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package Occurances;
+
+
+interface printable {
```
Comment (rohitlamsonge66) [inline NewPrograms/interfaceJava.java]: Remove not required space.

File: NewPrograms/treeSet.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
+package Occurances;
+
+import java.util.*;
+
+public class treeSet {
+    public static void main(String[] args) {
+        TreeSet<String> al = new TreeSet<String>();
```
Comment (rohitlamsonge66) [inline NewPrograms/treeSet.java]: use meaningfull variable name.

File: NewPrograms/JavaExceptionExample.java (hunk 5 above)
Comment (tejashrighogare) [reply NewPrograms/JavaExceptionExample.java]: Removed

File: NewPrograms/QueueClass.java (hunk 6 above)
Comment (tejashrighogare) [reply NewPrograms/QueueClass.java]: Done Correction

File: NewPrograms/QueueClass.java (hunk 7 above)
Comment (tejashrighogare) [reply NewPrograms/QueueClass.java]: Removed

File: NewPrograms/interfaceJava.java (hunk 9 above)
Comment (tejashrighogare) [reply NewPrograms/interfaceJava.java]: Done

File: NewPrograms/treeSet.java (hunk 10 above)
Comment (tejashrighogare) [reply NewPrograms/treeSet.java]: used meaningfull variable name

File: NewPrograms/funOverloading.java (hunk 8 above)
Comment (tejashrighogare) [reply NewPrograms/funOverloading.java]: Removed

**Review by rohitlamsonge66 — APPROVED** [review]

**Pull request merged**